fdmprinting/archive_retrieve.py

```diff
@@ -19,7 +20,8 @@ def get_database_handle():
             task_id text,
             image BLOB,
             raw_json text,
-            owner text
+            owner text,
+            fail_reason integer
```

Comment:
What happens when the `print_status` table already exists?  Will the `fail_reason` field not get added?

Reply:
hrmm - I'll look into it.  worst case we have an upgrade script to add it.  Right now this is what I see out of the box:

```
Loop starting
Replicator 1 (3DP-00M-748 replicator-1.synshop.net) unchanged, no DB updates
Failed to save printer status data/printers.sqlite . Error is: table print_status has no column named fail_reason
Replicator A (3DP-00M-784 replicator-a.synshop.net) changed, wrote to DB
Failed to save printer status data/printers.sqlite . Error is: table print_status has no column named fail_reason
Replicator α (3DP-00M-451 replicator-alpha.synshop.net) changed, wrote to DB
Loop complete
```

Reply:
so after flailing about for a bit, I think the easiest is to add a new function called something like `def migration_add_fail_reason():` .  you an put this in `archive_retrieve.py` or in it's own `sqlite_migration1.py` file.  While it's kinda a pain, I think it makes it really easy to deploy with something like:

1. `git pull origin`
2. `python sqlite_migration1.py`
3. `rc-service discord-bot restart && rc-service printer-loop restart`

This way you don't have to faff about with manual sqlite calls and such.  It also makes the upgrade more of a sure thing which you can test in dev.